pkg/live/inventoryrg_test.go

```diff
@@ -240,3 +280,729 @@ func TestIsResourceGroupInventory(t *testing.T) {
+					Name:      "test-deployment-2",
+					Namespace: "foo",
+				},
+				{
+					GroupKind: schema.GroupKind{
+						Group: "apps",
+						Kind:  "Deployment",
+					},
+					Name:      "test-deployment-3",
+					Namespace: "foo",
+				},
+				{
+					GroupKind: schema.GroupKind{
+						Group: "apps",
+						Kind:  "Deployment",
+					},
+					Name:      "test-deployment-4",
+					Namespace: "foo",
+				},
+			},
+			resourceCount: 3,
+			existingRG: inventoryWithObjs([]object.ObjMetadata{
+				{
+					GroupKind: schema.GroupKind{
+						Group: "apps",
+						Kind:  "Deployment",
+					},
+					Name:      "test-deployment",
+					Namespace: "foo",
+				},
+				{
+					GroupKind: schema.GroupKind{
+						Group: "apps",
+						Kind:  "Deployment",
+					},
+					Name:      "test-deployment-1",
+					Namespace: "foo",
+				},
+			}),
+			expNumberRGObjects: 2,
+		},
+		"Add add 0 new objects to existing ResourceGroup with no sharding should add sharded label": {
+			objMetas: []object.ObjMetadata{
+				{
+					GroupKind: schema.GroupKind{
+						Group: "apps",
+						Kind:  "Deployment",
+					},
+					Name:      "test-deployment",
+					Namespace: "foo",
+				},
+			},
+			resourceCount: 0,
+			existingRG: inventoryWithObjs([]object.ObjMetadata{
+				{
+					GroupKind: schema.GroupKind{
+						Group: "apps",
+						Kind:  "Deployment",
+					},
+					Name:      "test-deployment",
+					Namespace: "foo",
+				},
+			}),
+			expNumberRGObjects: 1,
+		},
+	}
+
+	for tn, tc := range testCases {
+		t.Run(tn, func(T *testing.T) {
+			dc, mapper := fakeClient()
+
+			irg := InventoryResourceGroup{
+				inv:           inventoryObj,
+				objMetas:      tc.objMetas,
+				strategy:      inventory.NameStrategy,
+				resourceCount: tc.resourceCount,
+			}
+			namespacedClient, err := irg.getNamespacedClient(dc, mapper)
+			if err != nil {
+				t.Error(err)
+			}
+
+			if tc.existingRG != nil {
+				_, err := namespacedClient.Create(context.TODO(), tc.existingRG, v1.CreateOptions{})
+				if err != nil {
+					t.Errorf("unable to create existing RG: %v", err)
+				}
+			}
+
+			err = irg.Apply(dc, mapper, inventory.StatusPolicyNone)
+
+			if err != nil {
+				t.Error(err)
+			}
+
+			clusterRGList, err := namespacedClient.List(context.TODO(), v1.ListOptions{
+				LabelSelector: irg.shardedLabel(),
+			})
+			if err != nil {
+				t.Error(err)
+			}
+
+			// Check we applied the ResourceGroups correctly.
+			resourcegroups := clusterRGList.Items
+
+			if len(resourcegroups) != tc.expNumberRGObjects {
+				t.Errorf("expected %d ResourceGroup objects, but found %d %+v", tc.expNumberRGObjects, len(resourcegroups), clusterRGList)
+			}
+
+			managedResources := make(map[object.ObjMetadata]struct{}, len(irg.objMetas))
+			for _, objMeta := range tc.objMetas {
+				managedResources[objMeta] = struct{}{}
+			}
+			allResources := len(managedResources)
+			var rgObjects int
+
+			for _, rg := range resourcegroups {
+				objs, err := irg.readResourceGroupObjects(&rg)
```

Comment:
Since we need to use it to test with, should this be a public package function?
Won't we also need it for e2e tests?

Reply:
Good idea, exporting this helper function!